### ISSUE TITLE
Hyperliquid - Perps balance used & free not correct

### ISSUE THREAD

**Issue by gaardiolor** [ccxt/ccxt#23997]
### Operating System

N/A

### CCXT Version

4.4.16

### Description

Hello!

TL;DR: Hyperliquid orders and balance are a bit sketchy in the first few seconds after creating the order. In order to fix this, I think that, for perps, it's best to set balance:
`['total']` to `['marginSummary']['accountValue']`
`['used']` to `['marginSummary']['accountValue'] - ['withdrawable']`
`['free']` to `['withdrawable']`

Balance before the order:
`2024-10-15 11:05:23,323 - freqtrade.exchange.exchange - INFO - API get_balances: {'info': {'marginSummary': {'accountValue': '51.468606', 'totalNtlPos': '0.0', 'totalRawUsd': '51.468606', 'totalMarginUsed': '0.0'}, 'crossMarginSummary': {'accountValue': '51.468606', 'totalNtlPos': '0.0', 'totalRawUsd': '51.468606', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '51.468606', 'assetPositions': [], 'time': '1728990323177'}, 'USDC': {'total':51.468606, 'used': 0.0, 'free': 51.468606}, 'timestamp': 1728990323177, 'datetime': '2024-10-15T11:05:23.177Z', 'free': {'USDC': 51.468606}, 'used': {'USDC': 0.0}, 'total': {'USDC': 51.468606}}`

Sometimes, when opening an order on hyperliquid, something strange happens. The order is 'resting' for a couple of seconds:
`2024-10-15 11:05:26,106 - freqtrade.exchange.exchange - INFO - API create_order: {'info': {'resting': {'oid': '41937919369'}}, 'id': '41937919369', 'clientOrderId': None, 'timestamp': None, 'datetime': None, 'lastTradeTimestamp': None, 'lastUpdateTimestamp': None, 'symbol': None, 'type': 'limit', 'timeInForce': None, 'postOnly': None, 'reduceOnly': None, 'side': None, 'price': None, 'triggerPrice': None, 'amount': None, 'cost': None, 'average': None, 'filled': None, 'remaining': None, 'status': 'open', 'fee': None, 'trades': [], 'fees': [], 'stopPrice': None, 'takeProfitPrice': None, 'stopLossPrice': None}`

The balance gets reported like this when that happens (notice ['USDC']['used']: 0.0 and ['USDC']['free'] being too high):
`2024-10-15 11:05:26,860 - freqtrade.exchange.exchange - INFO - API get_balances: {'info': {'marginSummary': {'accountValue': '51.468606', 'totalNtlPos': '0.0', 'totalRawUsd': '51.468606', 'totalMarginUsed': '0.0'}, 'crossMarginSummary': {'accountValue': '51.468606', 'totalNtlPos': '0.0', 'totalRawUsd': '51.468606', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '26.026361', 'assetPositions': [], 'time': '1728990326583'}, 'USDC': {'total': 51.468606, 'used': 0.0, 'free': 51.468606}, 'timestamp': 1728990326583, 'datetime': '2024-10-15T11:05:26.583Z', 'free': {'USDC': 51.468606}, 'used': {'USDC': 0.0}, 'total': {'USDC': 51.468606}}`

So everything is reported the same by the hyperliquid API, except "withdrawable". this was 51.468606 and dropped to 26.026361 , difference of 25.442245, which is about the same as the order size. 

a few seconds later, the order looks better:
`2024-10-15 11:05:28,917 - freqtrade.exchange.exchange - INFO - API fetch_order: {'info': {'order': {'coin': 'DOGE', 'side': 'B', 'limitPx': '0.11591', 'sz': '439.0', 'oid': '41937919369', 'timestamp': '1728990325980', 'triggerCondition': 'N/A', 'isTrigger': False, 'triggerPx': '0.0', 'children': [], 'isPositionTpsl': False, 'reduceOnly': False, 'orderType': 'Limit', 'origSz': '439.0', 'tif': 'Gtc', 'cloid': None}, 'status': 'open', 'statusTimestamp': '1728990325980'}, 'id': '41937919369', 'clientOrderId': None, 'timestamp': 1728990325980, 'datetime': '2024-10-15T11:05:25.980Z', 'lastTradeTimestamp': None, 'lastUpdateTimestamp': None, 'symbol': 'DOGE/USDC:USDC', 'type': 'limit', 'timeInForce': 'GTC', 'postOnly': False, 'reduceOnly': False, 'side': 'buy', 'price': 0.11591, 'triggerPrice': None, 'amount': 439.0, 'cost': 0.0, 'average': None, 'filled': 0.0, 'remaining': 439.0, 'status': 'open', 'fee': None, 'trades': [], 'fees': [], 'stopPrice': None, 'takeProfitPrice': None, 'stopLossPrice': None}`

Balance however still reports ['USDC']['used']: 0.0 and ['USDC']['free'] being too high :
`2024-10-15 11:05:32,713 - freqtrade.exchange.exchange - INFO - API get_balances: {'info': {'marginSummary': {'accountValue': '51.468606', 'totalNtlPos': '0.0', 'totalRawUsd': '51.468606', 'totalMarginUsed': '0.0'}, 'crossMarginSummary': {'accountValue': '51.468606', 'totalNtlPos': '0.0', 'totalRawUsd': '51.468606', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '26.026361', 'assetPositions': [], 'time': '1728990332486'}, 'USDC': {'total': 51.468606, 'used': 0.0, 'free': 51.468606}, 'timestamp': 1728990332486, 'datetime': '2024-10-15T11:05:32.486Z', 'free': {'USDC': 51.468606}, 'used': {'USDC': 0.0}, 'total': {'USDC': 51.468606}}`

Now when that happens, freqtrade (and other scripts) think it can open a larger position than possible, because ['USDC']['used']: 0.0 and ['USDC']['free'] being too high . Sometimes this fails because the first order just went through, sometimes it succeeds and as a consequence the first order is only filled partially. In the end, balance is reported:

`2024-10-15 11:31:08,133 - freqtrade.exchange.exchange - INFO - API get_balances: {'info': {'marginSummary': {'accountValue': '50.938961', 'totalNtlPos': '102.39894', 'totalRawUsd': '-51.459979', 'totalMarginUsed': '50.920635'}, 'crossMarginSummary': {'accountValue': '0.018326', 'totalNtlPos': '0.0', 'totalRawUsd': '0.018326', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '0.018326', 'assetPositions': [{'type': 'oneWay', 'position': {'coin': 'BTC', 'szi': '0.00115', 'leverage': {'type': 'isolated', 'value': '2', 'rawUsd': '-37.79819'}, 'entryPx': '65691.0', 'positionValue': '75.4147', 'unrealizedPnl': '-0.12995', 'returnOnEquity': '-0.00344035', 'liquidationPx': '33199.99121651', 'marginUsed': '37.61651', 'maxLeverage': '50', 'cumFunding': {'allTime': '-0.000183', 'sinceOpen': '0.0', 'sinceChange': '0.0'}}}, {'type': 'oneWay', 'position': {'coin': 'DOGE', 'szi': '236.0', 'leverage': {'type': 'isolated', 'value': '2', 'rawUsd': '-13.680115'}, 'entryPx': '0.11591', 'positionValue': '26.98424', 'unrealizedPnl': '-0.37052', 'returnOnEquity': '-0.02708998', 'liquidationPx': '0.05945291', 'marginUsed': '13.304125', 'maxLeverage': '20', 'cumFunding': {'allTime': '0.0', 'sinceOpen': '0.0', 'sinceChange': '0.0'}}}], 'time': '1728991867895'}, 'USDC': {'total': 50.938961, 'used': 50.920635, 'free': 0.018326}, 'timestamp': 1728991867895, 'datetime': '2024-10-15T11:31:07.895Z', 'free': {'USDC': 0.018326}, 'used': {'USDC': 50.920635}, 'total': {'USDC': 50.938961}}`

So, hyperliquid orders and balance are a bit sketchy in the first few seconds after creating the order. In order to fix this, I think that, for perps, it's best to set:
`['total']` to `['marginSummary']['accountValue']`
`['used']` to `['marginSummary']['accountValue'] - ['withdrawable']`
`['free']` to `['withdrawable']`

Can this be changed? Thanks!

**Comment by sc0Vu** [comment]
@gaardiolor Thanks for the report, I'll look into this issue.

After testing, I couldn’t reproduce this issue on the testnet (the order disappeared, but the withdrawable amount didn’t change, and the margin was taken, but the balance didn’t change either.). The testnet is behaving strangely. I’ll check on the mainnet.

**Comment by gaardiolor** [comment]
Hi @sc0Vu,

I did some testing myself, outside of freqtrade, just using ccxt. It seems the problem is a bit different than I thought.

- All orders I do are 'resting' for a moment, not some of them. Not sure why freqtrade reports only some of them as 'resting'.
- The orders, if they are open (not filled) do not update the 'used', wether they are resting or not. Used only goes up when the order gets filled.

```
    # Prep ccxt
    e = ccxt.hyperliquid()
    e.walletAddress = config['hyperliquid']['walletAddress']
    e.privateKey = config['hyperliquid']['privateKey']
    e.aiohttp_proxy = 'http://127.0.0.1:8080'

    # Prep loop
    pair = 'BTC/USDC:USDC'
    await e.set_leverage(symbol=pair, leverage=2)
    balance = await e.fetch_balance()
    logging.info(f"Starting Balance USDC: {balance['USDC']} Balance: {balance}")

    order = await e.create_order(pair, 'limit', 'buy', 0.0005, 65000, {'timeInForce': 'GTC'})
    logging.info(f"Created order: {order}")

    while True:
        balance = await e.fetch_balance()
        logging.info(f"Balance USDC: {balance['USDC']} Balance: {balance}")
        logging.info(f"Orders: {await e.fetch_open_orders()}")
        if balance['USDC']['used'] != 0.0:
            break
        await asyncio.sleep(1)

    logging.info(f"Balance used went up, exiting")
    await e.close()
```


```
[2024-10-16 21:25:06] - Starting Balance USDC: {'total': 51.64387, 'used': 0.0, 'free': 51.64387} Balance: {'info': {'marginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMarginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '51.64387', 'assetPositions': [], 'time': '1729106702422'}, 'USDC': {'total': 51.64387, 'used': 0.0, 'free': 51.64387}, 'timestamp': 1729106702422, 'datetime': '2024-10-16T19:25:02.422Z', 'free': {'USDC': 51.64387}, 'used': {'USDC': 0.0}, 'total': {'USDC': 51.64387}}
[2024-10-16 21:25:08] - Created order: {'info': {'resting': {'oid': '42211832946'}}, 'id': '42211832946', 'clientOrderId': None, 'timestamp': None, 'datetime': None, 'lastTradeTimestamp': None, 'lastUpdateTimestamp': None, 'symbol': None, 'type': None, 'timeInForce': None, 'postOnly': None, 'reduceOnly': None, 'side': None, 'price': None, 'triggerPrice': None, 'amount': None, 'cost': None, 'average': None, 'filled': None, 'remaining': None, 'status': None, 'fee': None, 'trades': [], 'fees': [], 'stopPrice': None, 'takeProfitPrice': None, 'stopLossPrice': None}
[2024-10-16 21:25:08] - Balance USDC: {'total': 51.64387, 'used': 0.0, 'free': 51.64387} Balance: {'info': {'marginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMarginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '35.39387', 'assetPositions': [], 'time': '1729106703947'}, 'USDC': {'total': 51.64387, 'used': 0.0, 'free': 51.64387}, 'timestamp': 1729106703947, 'datetime': '2024-10-16T19:25:03.947Z', 'free': {'USDC': 51.64387}, 'used': {'USDC': 0.0}, 'total': {'USDC': 51.64387}}
[2024-10-16 21:25:08] - Orders: [{'info': {'coin': 'BTC', 'side': 'B', 'limitPx': '65000.0', 'sz': '0.0005', 'oid': '42211832946', 'timestamp': '1729106703746', 'triggerCondition': 'N/A', 'isTrigger': False, 'triggerPx': '0.0', 'children': [], 'isPositionTpsl': False, 'reduceOnly': False, 'orderType': 'Limit', 'origSz': '0.0005', 'tif': 'Gtc', 'cloid': None}, 'id': '42211832946', 'clientOrderId': None, 'timestamp': 1729106703746, 'datetime': '2024-10-16T19:25:03.746Z', 'lastTradeTimestamp': None, 'lastUpdateTimestamp': None, 'symbol': 'BTC/USDC:USDC', 'type': 'limit', 'timeInForce': 'GTC', 'postOnly': False, 'reduceOnly': False, 'side': 'buy', 'price': 65000.0, 'triggerPrice': None, 'amount': 0.0005, 'cost': 0.0, 'average': None, 'filled': 0.0, 'remaining': 0.0005, 'status': 'open', 'fee': None, 'trades': [], 'fees': [], 'stopPrice': None, 'takeProfitPrice': None, 'stopLossPrice': None}]
[2024-10-16 21:25:10] - Balance USDC: {'total': 51.64387, 'used': 0.0, 'free': 51.64387} Balance: {'info': {'marginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMarginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '35.39387', 'assetPositions': [], 'time': '1729106705668'}, 'USDC': {'total': 51.64387, 'used': 0.0, 'free': 51.64387}, 'timestamp': 1729106705668, 'datetime': '2024-10-16T19:25:05.668Z', 'free': {'USDC': 51.64387}, 'used': {'USDC': 0.0}, 'total': {'USDC': 51.64387}}
[2024-10-16 21:25:10] - Orders: [{'info': {'coin': 'BTC', 'side': 'B', 'limitPx': '65000.0', 'sz': '0.0005', 'oid': '42211832946', 'timestamp': '1729106703746', 'triggerCondition': 'N/A', 'isTrigger': False, 'triggerPx': '0.0', 'children': [], 'isPositionTpsl': False, 'reduceOnly': False, 'orderType': 'Limit', 'origSz': '0.0005', 'tif': 'Gtc', 'cloid': None}, 'id': '42211832946', 'clientOrderId': None, 'timestamp': 1729106703746, 'datetime': '2024-10-16T19:25:03.746Z', 'lastTradeTimestamp': None, 'lastUpdateTimestamp': None, 'symbol': 'BTC/USDC:USDC', 'type': 'limit', 'timeInForce': 'GTC', 'postOnly': False, 'reduceOnly': False, 'side': 'buy', 'price': 65000.0, 'triggerPrice': None, 'amount': 0.0005, 'cost': 0.0, 'average': None, 'filled': 0.0, 'remaining': 0.0005, 'status': 'open', 'fee': None, 'trades': [], 'fees': [], 'stopPrice': None, 'takeProfitPrice': None, 'stopLossPrice': None}]
[2024-10-16 21:25:11] - Balance USDC: {'total': 51.64387, 'used': 0.0, 'free': 51.64387} Balance: {'info': {'marginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMarginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '35.39387', 'assetPositions': [], 'time': '1729106707167'}, 'USDC': {'total': 51.64387, 'used': 0.0, 'free': 51.64387}, 'timestamp': 1729106707167, 'datetime': '2024-10-16T19:25:07.167Z', 'free': {'USDC': 51.64387}, 'used': {'USDC': 0.0}, 'total': {'USDC': 51.64387}}
[2024-10-16 21:25:12] - Orders: [{'info': {'coin': 'BTC', 'side': 'B', 'limitPx': '65000.0', 'sz': '0.0005', 'oid': '42211832946', 'timestamp': '1729106703746', 'triggerCondition': 'N/A', 'isTrigger': False, 'triggerPx': '0.0', 'children': [], 'isPositionTpsl': False, 'reduceOnly': False, 'orderType': 'Limit', 'origSz': '0.0005', 'tif': 'Gtc', 'cloid': None}, 'id': '42211832946', 'clientOrderId': None, 'timestamp': 1729106703746, 'datetime': '2024-10-16T19:25:03.746Z', 'lastTradeTimestamp': None, 'lastUpdateTimestamp': None, 'symbol': 'BTC/USDC:USDC', 'type': 'limit', 'timeInForce': 'GTC', 'postOnly': False, 'reduceOnly': False, 'side': 'buy', 'price': 65000.0, 'triggerPrice': None, 'amount': 0.0005, 'cost': 0.0, 'average': None, 'filled': 0.0, 'remaining': 0.0005, 'status': 'open', 'fee': None, 'trades': [], 'fees': [], 'stopPrice': None, 'takeProfitPrice': None, 'stopLossPrice': None}]
[2024-10-16 21:25:13] - Balance USDC: {'total': 51.64387, 'used': 0.0, 'free': 51.64387} Balance: {'info': {'marginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMarginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '35.39387', 'assetPositions': [], 'time': '1729106708775'}, 'USDC': {'total': 51.64387, 'used': 0.0, 'free': 51.64387}, 'timestamp': 1729106708775, 'datetime': '2024-10-16T19:25:08.775Z', 'free': {'USDC': 51.64387}, 'used': {'USDC': 0.0}, 'total': {'USDC': 51.64387}}
[2024-10-16 21:25:13] - Orders: [{'info': {'coin': 'BTC', 'side': 'B', 'limitPx': '65000.0', 'sz': '0.0005', 'oid': '42211832946', 'timestamp': '1729106703746', 'triggerCondition': 'N/A', 'isTrigger': False, 'triggerPx': '0.0', 'children': [], 'isPositionTpsl': False, 'reduceOnly': False, 'orderType': 'Limit', 'origSz': '0.0005', 'tif': 'Gtc', 'cloid': None}, 'id': '42211832946', 'clientOrderId': None, 'timestamp': 1729106703746, 'datetime': '2024-10-16T19:25:03.746Z', 'lastTradeTimestamp': None, 'lastUpdateTimestamp': None, 'symbol': 'BTC/USDC:USDC', 'type': 'limit', 'timeInForce': 'GTC', 'postOnly': False, 'reduceOnly': False, 'side': 'buy', 'price': 65000.0, 'triggerPrice': None, 'amount': 0.0005, 'cost': 0.0, 'average': None, 'filled': 0.0, 'remaining': 0.0005, 'status': 'open', 'fee': None, 'trades': [], 'fees': [], 'stopPrice': None, 'takeProfitPrice': None, 'stopLossPrice': None}]
[2024-10-16 21:25:14] - Balance USDC: {'total': 51.64387, 'used': 0.0, 'free': 51.64387} Balance: {'info': {'marginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMarginSummary': {'accountValue': '51.64387', 'totalNtlPos': '0.0', 'totalRawUsd': '51.64387', 'totalMarginUsed': '0.0'}, 'crossMaintenanceMarginUsed': '0.0', 'withdrawable': '35.39387', 'assetPositions': [], 'time': '1729106710360'}, 'USDC': {'total': 51.64387, 'used': 0.0, 'free': 51.64387}, 'timestamp': 1729106710360, 'datetime': '2024-10-16T19:25:10.360Z', 'free': {'USDC': 51.64387}, 'used': {'USDC': 0.0}, 'total': {'USDC': 51.64387}}
[2024-10-16 21:25:15] - Orders: [{'info': {'coin': 'BTC', 'side': 'B', 'limitPx': '65000.0', 'sz': '0.0005', 'oid': '42211832946', 'timestamp': '1729106703746', 'triggerCondition': 'N/A', 'isTrigger': False, 'triggerPx': '0.0', 'children': [], 'isPositionTpsl': False, 'reduceOnly': False, 'orderType': 'Limit', 'origSz': '0.0005', 'tif': 'Gtc', 'cloid': None}, 'id': '42211832946', 'clientOrderId': None, 'timestamp': 1729106703746, 'datetime': '2024-10-16T19:25:03.746Z', 'lastTradeTimestamp': None, 'lastUpdateTimestamp': None, 'symbol': 'BTC/USDC:USDC', 'type': 'limit', 'timeInForce': 'GTC', 'postOnly': False, 'reduceOnly': False, 'side': 'buy', 'price': 65000.0, 'triggerPrice': None, 'amount': 0.0005, 'cost': 0.0, 'average': None, 'filled': 0.0, 'remaining': 0.0005, 'status': 'open', 'fee': None, 'trades': [], 'fees': [], 'stopPrice': None, 'takeProfitPrice': None, 'stopLossPrice': None}]
```

Now in my previous tests (in freqtrade), I was under the impression that I was using isolated margin, but I just figured out that CCXT doesn't set that correctly. Maybe that's the cause of my issue instead of the balance. I'll log a new issue for that. 

Not sure if the behaviour described in this issue is working as intended or not..

**Comment by gaardiolor** [comment]
Did another test, comparing with gate.io. gate.io balance 'used' does go up when there's an open swap order, hyperliquid does not:

Creating an order with a too low price, so it remains open:
Gate:
```
    # Prep ccxt
    e = ccxt.gate()
    e.options['defaultType'] = 'swap'
    e.apiKey = config['gate']['apiKey']
    e.secret = config['gate']['secret']
    e.aiohttp_proxy = 'http://127.0.0.1:8080'

    logging.info(f"Open orders: {len(await e.fetch_open_orders())}")
    logging.info(f"Starting balance USDT: {(await e.fetch_balance())['USDT']}")

    # Each contract is 10 DOGE. Price set too low so the order remains open
    await e.create_order('DOGE/USDT:USDT', 'limit', 'buy', 15, 0.1, {'timeInForce': 'GTC', 'leverage': 2, 'side': 'buy', 'reduceOnly': False})

    await asyncio.sleep(1)

    logging.info(f"Open orders: {len(await e.fetch_open_orders())}")
    logging.info(f"Balance USDT: {(await e.fetch_balance())['USDT']}")

    for order in await e.fetch_open_orders():
        logging.info(f"Canceling {order['id']} {order['symbol']}")
        await e.cancel_order(order['id'], order['symbol'])
    await e.close()
--
Open orders: 0
Starting balance USDT: {'free': 47.76887671663, 'used': 0.0, 'total': 47.76887671663}
Open orders: 1
Balance USDT: {'free': 40.24637671663, 'used': 7.5225, 'total': 47.76887671663}   # Used goes up
Canceling 543428705289 DOGE/USDT:USDT
```

Hyperliquid:
```
    # Prep ccxt
    e = ccxt.hyperliquid()
    e.walletAddress = config['hyperliquid']['walletAddress']
    e.privateKey = config['hyperliquid']['privateKey']

    e.aiohttp_proxy = 'http://127.0.0.1:8080'

    logging.info(f"Open orders: {len(await e.fetch_open_orders())}")
    logging.info(f"Starting balance USDT: {(await e.fetch_balance())['USDC']}")

    await e.set_margin_mode('isolated', 'DOGE/USDC:USDC', params={"leverage": 2})
    await e.create_order('DOGE/USDC:USDC', 'limit', 'buy', 150, 0.1, {'timeInForce': 'GTC'})

    await asyncio.sleep(1)

    logging.info(f"Open orders: {len(await e.fetch_open_orders())}")
    logging.info(f"Balance USDT: {(await e.fetch_balance())['USDC']}")

    for order in await e.fetch_open_orders():
        logging.info(f"Canceling {order['id']} {order['symbol']}")
        await e.cancel_order(order['id'], order['symbol'])
    await e.close()
--
Open orders: 0
Starting balance USDT: {'total': 51.736159, 'used': 0.0, 'free': 51.736159}
Open orders: 1
Balance USDT: {'total': 51.736159, 'used': 0.0, 'free': 51.736159}  # Used does not go up - has to be fixed
Canceling 42284531133 DOGE/USDC:USDC
```

So 'resting' has nothing to do with it.. the problem with hyperliquid is that open orders are not reflected in ['USDC']['used']

**Comment by gaardiolor** [comment]
Hi @sc0Vu , did some more testing, in my environment I've changed [this line](https://github.com/ccxt/ccxt/blob/0206b8184e7258ccc4817593565d56f16cb72ed9/python/ccxt/async_support/hyperliquid.py#L652) to `'free': self.safe_float(response, 'withdrawable')` . Everything seems to work now. Hope you can fix it?

(Maybe also [here](https://github.com/ccxt/ccxt/blob/0206b8184e7258ccc4817593565d56f16cb72ed9/python/ccxt/async_support/hyperliquid.py#L642) and [here](https://github.com/ccxt/ccxt/blob/0206b8184e7258ccc4817593565d56f16cb72ed9/python/ccxt/async_support/hyperliquid.py#L644) change the variable name `free` to `used` ? seems like a typo)
